functors task (2)/student/src/impl.cpp

```diff
@@ -7,48 +7,44 @@ void DataBrowser::userLeave(const std::string &userId)
 
 bool DataBrowser::getDataType1(const std::string &userId, std::vector<size_t> &returnValues) const
 {
-    const auto& it = m_dataReaders.find(userId);
-    if (it == m_dataReaders.cend())
+    auto func = [this, & returnValues](const std::unique_ptr<IDataSelector>& selector)
     {
-        return false;
-    }
-    if (it->second == nullptr)
-    {
-        return false;
-    }
-    return it->second->getDataType1(returnValues, 0);
+        auto f = [](const IDataSelector* selector, std::vector<size_t> &returnValues)
```

Comment:
_Such names are acceptable for templated code, but in future consider more descriptive names_